demos/induction_heads_multigpu.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import annotations
+
+import argparse
+import os
+from argparse import Namespace
+from functools import partial
+from typing import Any, Callable
+
+import einops
+import matplotlib.pyplot as plt
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+from torch import nn
+from torch.distributed.fsdp import (
+    FullyShardedDataParallel as FSDP,
+)
+from torch.distributed.fsdp import (
+    MixedPrecision,
+    ShardingStrategy,
+)
+from torch.distributed.fsdp.wrap import (
+    transformer_auto_wrap_policy,
+)
+from transformers import (
+    LlamaConfig,
+    LlamaForCausalLM,
+)
+from transformers.models.llama.modeling_llama import LlamaDecoderLayer
+
+from flex_model.core import FlexModel, HookFunction
+
+
+def setup() -> None:
+    """Instantiate process group."""
+    dist.init_process_group("nccl")
+
+
+def cleanup() -> None:
+    """Destroy process group."""
+    dist.destroy_process_group()
+
+
+def args() -> Namespace:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_path", type=str, required=True)
+    return parser.parse_args()
+
+
+def setup_model(model_path: str, rank: int) -> tuple[nn.Module, LlamaConfig]:
```

Comment:
Minor: Signature would be cleaner without the rank arg, which can just be obtained via `dist.get_rank()` or other

Reply:
Turned this into a global